crates/router/src/core/webhooks.rs

```diff
@@ -90,6 +90,99 @@ async fn payments_incoming_webhook_flow(
     Ok(())
 }
 
+fn get_refund_status_from_event_type(
+    event_type: api_models::webhooks::IncomingWebhookEvent,
+) -> CustomResult<storage_models::enums::RefundStatus, errors::WebhooksFlowError> {
+    match event_type {
+        api_models::webhooks::IncomingWebhookEvent::RefundSuccess => {
+            Ok(storage_models::enums::RefundStatus::Success)
+        }
+        api_models::webhooks::IncomingWebhookEvent::RefundFailure => {
+            Ok(storage_models::enums::RefundStatus::Failure)
+        }
+        _ => Err(errors::WebhooksFlowError::RefundsCoreFailed.into()),
+    }
+}
+
+#[instrument(skip_all)]
+async fn refunds_incoming_webhook_flow(
+    state: AppState,
+    merchant_account: storage::MerchantAccount,
+    webhook_details: api::IncomingWebhookDetails,
+    connector_name: &str,
+    source_verified: bool,
+    event_type: api_models::webhooks::IncomingWebhookEvent,
+) -> CustomResult<(), errors::WebhooksFlowError> {
+    let db = &*state.store;
+    //find refund by connector refund id
+    let refund = db
+        .find_refund_by_merchant_id_connector_refund_id_connector(
+            &merchant_account.merchant_id,
+            &webhook_details.object_reference_id,
+            connector_name,
+            merchant_account.storage_scheme,
+        )
+        .await
+        .change_context(errors::WebhooksFlowError::ResourceNotFound)
+        .attach_printable_lazy(|| "Failed fetching the refund")?;
+    let refund_id = refund.refund_id.to_owned();
+    //if source verified then update refund status else trigger refund sync
+    let updated_refund = if source_verified {
+        let refund_update = storage::RefundUpdate::StatusUpdate {
+            connector_refund_id: None,
+            sent_to_gateway: true,
+            refund_status: get_refund_status_from_event_type(event_type)?,
+        };
+        state
+            .store
+            .update_refund(
+                refund.to_owned(),
+                refund_update,
+                merchant_account.storage_scheme,
+            )
+            .await
+            .change_context(errors::WebhooksFlowError::RefundsCoreFailed)
+            .attach_printable_lazy(|| {
+                format!(
+                    "Failed while updating refund: refund_id: {}",
+                    refund.refund_id
+                )
+            })?
+    } else {
+        refunds::refund_retrieve_core(&state, merchant_account.clone(), refund_id)
+            .await
+            .change_context(errors::WebhooksFlowError::RefundsCoreFailed)
+            .attach_printable_lazy(|| {
+                format!(
+                    "Failed while updating refund: refund_id: {}",
+                    refund.refund_id
+                )
+            })?
+    };
+    let refund_id = updated_refund.refund_id.clone();
```

Comment:
The refund ID for the updated refund must be the same as before. This reassignment is not necessary.

Reply:
Yes, will remove that!